NoDenotation.owner in LabmdaLift

```
class KillLambdaLift {
  private def com: this.type = _root_.scala.Predef.???
}
```
```
exception while transforming _root_ of class class dotty.tools.dotc.ast.Trees$Ident # 31
exception while transforming _root_.scala of class class dotty.tools.dotc.ast.Trees$Select # 32
exception while transforming _root_.scala.Predef of class class dotty.tools.dotc.ast.Trees$Select # 105
exception while transforming _root_.scala.Predef.??? of class class dotty.tools.dotc.ast.Trees$Select # 394
exception while transforming _root_.scala.Predef.???() of class class dotty.tools.dotc.ast.Trees$Apply # 395
exception while transforming private def com(): i321 = _root_.scala.Predef.???() of class class dotty.tools.dotc.ast.Trees$DefDef # 396
exception while transforming  extends Object {
  def <init>(): i321 = {
    super()
    ()
  }
  private def com(): i321 = _root_.scala.Predef.???()
} of class class dotty.tools.dotc.ast.Trees$Template # 425
exception while transforming class i321 extends Object {
  def <init>(): i321 = {
    super()
    ()
  }
  private def com(): i321 = _root_.scala.Predef.???()
} of class class dotty.tools.dotc.ast.Trees$TypeDef # 426
exception while transforming package <empty> {
  class i321 extends Object {
    def <init>(): i321 = {
      super()
      ()
    }
    private def com(): i321 = _root_.scala.Predef.???()
  }
  final lazy module val i321: i321$ = new i321$()
  final module class i321$ extends Object { this: i321$ =>
    def <init>(): i321$ = {
      super()
      ()
    }
  }
} of class class dotty.tools.dotc.ast.Trees$PackageDef # 438
Exception in thread "main" java.lang.AssertionError: NoDenotation.owner
	at dotty.tools.dotc.core.SymDenotations$NoDenotation$.owner(SymDenotations.scala:1575)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.enclosingMethod(SymDenotations.scala:747)
	at dotty.tools.dotc.transform.SymUtils$.enclosure$extension(SymUtils.scala:68)
	at dotty.tools.dotc.transform.LambdaLift$LambdaLifter.transformIdent(LambdaLift.scala:375)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.goIdent(TreeTransform.scala:563)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformNamed(TreeTransform.scala:958)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1205)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformNamed(TreeTransform.scala:963)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1205)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformNamed(TreeTransform.scala:963)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1205)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformNamed(TreeTransform.scala:963)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1205)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformUnnamed(TreeTransform.scala:1028)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1206)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformNamed(TreeTransform.scala:997)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1205)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.dotty$tools$dotc$transform$TreeTransforms$TreeTransformer$$transformStat$1(TreeTransform.scala:1231)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$38.apply(TreeTransform.scala:1235)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$38.apply(TreeTransform.scala:1235)
	at dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:50)
	at dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:66)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformStats(TreeTransform.scala:1235)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformUnnamed(TreeTransform.scala:1176)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1206)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformNamed(TreeTransform.scala:1004)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1205)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.dotty$tools$dotc$transform$TreeTransforms$TreeTransformer$$transformStat$1(TreeTransform.scala:1231)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$38.apply(TreeTransform.scala:1235)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$38.apply(TreeTransform.scala:1235)
	at dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:50)
	at dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:66)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformStats(TreeTransform.scala:1235)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transformUnnamed(TreeTransform.scala:1185)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1206)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer$$anonfun$transform$2.apply(TreeTransform.scala:1198)
	at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:147)
	at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:51)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.transform(TreeTransform.scala:1197)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.macroTransform(TreeTransform.scala:556)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.run(TreeTransform.scala:480)
	at dotty.tools.dotc.core.Phases$Phase$$anonfun$runOn$1.apply(Phases.scala:265)
	at dotty.tools.dotc.core.Phases$Phase$$anonfun$runOn$1.apply(Phases.scala:263)
	at scala.collection.immutable.List.map(List.scala:273)
	at dotty.tools.dotc.core.Phases$Phase$class.runOn(Phases.scala:263)
	at dotty.tools.dotc.transform.TreeTransforms$TreeTransformer.runOn(TreeTransform.scala:474)
	at dotty.tools.dotc.Run$$anonfun$compileSources$1$$anonfun$apply$mcV$sp$3.apply(Run.scala:50)
	at dotty.tools.dotc.Run$$anonfun$compileSources$1$$anonfun$apply$mcV$sp$3.apply(Run.scala:47)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at dotty.tools.dotc.Run$$anonfun$compileSources$1.apply$mcV$sp(Run.scala:47)
	at dotty.tools.dotc.Run$$anonfun$compileSources$1.apply(Run.scala:42)
	at dotty.tools.dotc.Run$$anonfun$compileSources$1.apply(Run.scala:42)
	at dotty.tools.dotc.util.Stats$.monitorHeartBeat(Stats.scala:68)
	at dotty.tools.dotc.Run.compileSources(Run.scala:41)
	at dotty.tools.dotc.Run.compile(Run.scala:32)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:20)
	at dotty.tools.dotc.Main$.doCompile(Main.scala:26)
	at dotty.tools.dotc.Driver.process(Driver.scala:31)
	at dotty.tools.dotc.Driver.main(Driver.scala:40)
	at dotty.tools.dotc.Main.main(Main.scala)
````

--- a/src/dotty/tools/dotc/core/SymDenotations.scala
+++ b/src/dotty/tools/dotc/core/SymDenotations.scala
@@ -744,7 +744,8 @@ object SymDenotations {
     final def enclosingMethod(implicit ctx: Context): Symbol =
       if (this is (Method, butNot = Label)) symbol
       else if (this.isClass) primaryConstructor
-      else owner.enclosingMethod
+      else if (this.exists) owner.enclosingMethod
+      else NoSymbol
 
     /** The top-level class containing this denotation,
      *  except for a toplevel module, where its module class is returned.

--- a/tests/pos/KillLambdaLift.scala
+++ b/tests/pos/KillLambdaLift.scala
@@ -1,0 +1,3 @@
+class KillLambdaLift {
+  private def com: this.type = _root_.scala.Predef.???
+}

